src/test/java/eu/europeana/metis/sandbox/service/workflow/TransformationServiceImplTest.java

```diff
@@ -1,143 +1,93 @@
-        .thenReturn(expected.getBytes());
-
-    var result = service.transform("identifier", xsltContent, recordContent);
-
-    assertArrayEquals(expected.getBytes(), result);
-  }
+  TransformationServiceImpl transformationService;
 
   @Test
-  void transformToEdmInternal_expectSuccess() throws IOException, TransformationException {
-    var input = testUtils.readFileToBytes("record"+ File.separator+"transform"+File.separator+"record-input.xml");
-    var expected = testUtils.readFileToString("record"+File.separator+"transform"+File.separator+"record-expected.xml");
-
-    var record = Record.builder()
-        .datasetId("1").datasetName("").country(Country.ITALY).language(Language.IT).content(input)
-        .recordId("").build();
-
-    when(objectProvider.getObject(anyString(), anyString(), anyString()))
-        .thenReturn(xsltTransformer);
-    when(xsltTransformer.transformToBytes(any(byte[].class), any(EuropeanaGeneratedIdsMap.class)))
-        .thenReturn(expected.getBytes());
+  void transformationService_expectSuccess() throws IOException {
+
+    var input = testUtils.readFileToString(
+        "record" + File.separator + "transform" + File.separator + "record-input.xml");
+    var expected = testUtils.readFileToString(
+        "record" + File.separator + "transform" + File.separator + "record-expected.xml");
+    var transformFile = testUtils.readFileToString(
+        "record" + File.separator + "defaultTransform.xslt");
+
+    var inputRecord = Record.builder()
+        .datasetId("1").datasetName("One")
+        .country(Country.ITALY)
+        .language(Language.IT)
+        .content(input.getBytes(StandardCharsets.UTF_8))
+        .recordId("1").build();
 
-    var result = service.transformToEdmInternal(record);
+    var expectedRecord = Record.builder()
+        .datasetId("1").datasetName("One")
+        .country(Country.ITALY)
+        .language(Language.IT)
+        .content(expected.getBytes(StandardCharsets.UTF_8))
+        .recordId("1").build();
 
-    assertArrayEquals(expected.getBytes(), result.getRecord().getContent());
-  }
+    var expectedRecordInfo = new RecordInfo(expectedRecord, Collections.emptyList());
 
-  @Test
-  void transform_nullRecord_expectFail() throws IOException {
-    ByteArrayInputStream xsltContent = new ByteArrayInputStream(
-        Files.readAllBytes(Paths.get("src/test/resources/record/xsltFileContent.xsl")));
-    assertThrows(NullPointerException.class, () -> service.transform("identifier", xsltContent, null));
-  }
+    TransformXsltEntity transformXsltEntity = new TransformXsltEntity(transformFile);
+    when(transformXsltRepository.findById(anyInt())).thenReturn(Optional.of(transformXsltEntity));
 
-  @Test
-  void transform_nullXsltContent_expectFail() {
-    assertThrows(RecordProcessingException.class, () -> service.transform("identifier", null, "record".getBytes(
-        StandardCharsets.UTF_8)));
-  }
+    var result = transformationService.transform(inputRecord);
```

Comment:
This gives an error probably another method was meant here?

Reply:
Method name changed. It's fixed now